python/lsst/summit/utils/blockUtils.py

```diff
@@ -0,0 +1,438 @@
+        return (
+            f"dayObs: {self.dayObs}\n"
+            f"seqNum: {self.seqNum}\n"
+            f"blockNumber: {self.blockNumber}\n"
+            f"blockId: {self.blockId}\n"
+            f"begin: {self.begin.isot}\n"
+            f"end: {self.end.isot}\n"
+            f"salIndices: {self.salIndices}\n"
+            f"tickets: {self.tickets}\n"
+            f"states: \n{newline.join([str(state) for state in self.states])}"
+        )
+
+
+@dataclass(kw_only=True, frozen=True)
+class ScriptStatePoint:
+    """The execution state of a script at a point in time.
+
+    Parameters
+    ----------
+    time : `astropy.time.Time`
+        The time of the state change.
+    state : `lsst.summit.utils.enums.ScriptState`
+        The state of the script at this point in time.
+    reason : `str`
+        The reason for the state change, if given.
+    """
+    time: Time
+    state: ScriptState
+    reason: str
+
+    def __repr__(self):
+        return (
+            f"ScriptStatePoint(time={self.time!r}, state={self.state!r}, reason={self.reason!r})"
+        )
+
+    def _ipython_display_(self):
+        """This is the function which runs when someone executes a cell in a
+        notebook with just the class instance on its own, without calling
+        print() or str() on it.
+        """
+        print(self.__str__())
+
+    def __str__(self):
+        reasonStr = f" - {self.reason}" if self.reason else ""
+        return (f"{self.state.name:>10} @ {self.time.isot}{reasonStr}")
+
+
+class BlockParser:
+    """A class to parse BLOCK data from the EFD.
+
+    Information on executed blocks is stored in the EFD in the
+    ``lsst.sal.Script.logevent_state`` topic. This class parses that topic and
+    provides methods to get information on the blocks which were run on a given
+    dayObs. It also provides methods to get the events which occurred during a
+    given block, and also to get the block in which a specified event occurred,
+    if any.
+
+    Parameters
+    ----------
+    dayObs : `int`
+        The dayObs to get the block data for.
+    client : `lsst_efd_client.efd_client.EfdClient`, optional
+        The EFD client to use. If not specified, a new one is created.
+    """
+
+    def __init__(self, dayObs, client=None):
+        self.log = logging.getLogger("lsst.summit.utils.blockUtils.BlockParser")
+        self.dayObs = dayObs
+
+        self.client = client
+        if client is None:
+            self.client = makeEfdClient()
+
+        t0 = time.time()
+        self.getDataForDayObs()
+        self.log.debug(f"Getting data took {(time.time()-t0):.2f} seconds")
+        t0 = time.time()
+        self.augmentData()
+        self.log.debug(f"Parsing data took {(time.time()-t0):.5f} seconds")
+
+    def getDataForDayObs(self):
+        """Retrieve the data for the specified dayObs from the EFD.
+        """
+        data = getEfdData(self.client, 'lsst.sal.Script.logevent_state', dayObs=self.dayObs)
```

Comment:
In general, hard coding is obviously something to be very wary of, but as far as I know, this is the one and only place that the BLOCK info is recorded, and given this is inside of the `BlockParser`, that seems not just OK, but almost necessary. It's like if there was a utility function which was specifically for getting AuxTel elevation position, then it would be the right thing to do, in that case, to hard code the topic. I absolutely agree with the general point about hard coding strings though, but feel like in this case, it would just be confusing and add nothing (because nobody would ever change it, but it would suggest to them that they might then want to). @tribeiro can you just confirm that this is indeed the only place where I would ever need to look for TMA BLOCK info?